src/editorialPalette.ts

```diff
@@ -0,0 +1,102 @@
+// ----- Imports ----- //
+
+import {
+    text,
+    background,
+    neutral,
+    news,
+    opinion,
+    sport,
+    culture,
+    lifestyle,
+} from '@guardian/src-foundations/palette';
+import { Format, Design, Display, Pillar } from '@guardian/types/Format';
+
+
+// ----- Types ----- //
+
+interface Colour {
+    light: string;
+    dark: string;
+}
+
+interface Palette {
+    headlineFont: Colour;
```

Comment:
Minor: I might suggest `headlineText` for better consistency with Source.

Reply:
Good idea! Updated.

Reply:
It's worth spending a bit of time thinking about how the naming system will scale. I know @oliverlloyd has pointed to the potential problem with the `Colour` abstraction defining a `light` and a `dark` mode. This could become a bit verbose or restrictive for DCR that only has a light mode. There might be ways for DCR to abstract this away in their application, but it's worth bearing in mind.

I'll give you a peek at Source's colour token naming system:

```
[theme][property].[component][variant][state][mode]
```

- theme: the colour theme to which this token applies
- property: the style property for which this token is intended
- component: the design system component to which this token is applied
- variant: there are sometimes multiple variants of each token. For example, text in a component may be primary (core) or secondary (supporting)
- state: user interaction that may have affected the appearance, such as hover or checked
- mode: the colour mode (i.e. inverse represents dark mode)

Note that in most cases, not all of these parts will be present.

Examples:

```
text.primary
```

This is a text colour, for primary text. There is no theme specified, so we assume it’s for the default theme. Note that it’s not for a specific component. “Primary” is the variant, telling us it should be applied to the most important or core text. There is no state, so we assume there has been no user interaction. There is no mode, so we assume the default colour mode ("light mode")

```
brandBackground.cardArticleHover
```

It is a background colour on brand theme, for the article card, in the hover state.

Reply:
This is very thorough, thanks!

Two questions:

1. What does the dot signify in the hierarchy (I know it represents a method or property lookup, but why at that level)?
2. How would you define the **property** for text in a headline? Would it be `headlineText.primary`?

Reply:
The position of the dot was my best guess at where to split the palette to make it more treeshakable. It's likely that most applications won't be using multiple themes on one page, and those that used multiple themes might not want all the colours for each theme. I think you can ignore this consideration for your purpose.

In your example `text` is the property... it represents the CSS property to which this toekn will be applied (well, technically it's the CSS `color` property, but `color` is a bit vague!) 

I think `headlineText` is a good name. Abstracted it's `[component][property]` whereas Source goes with `[property][component]` but I don't think it matters too much.

There's only one headline per article, so you won't have variants like "primary headline" and "supporting headline".

It's worth taking a look at other design systems to see how they name their tokens. Then write down what your system is, so you can (attempt to!) keep it consistent:

- [Salesforce](https://www.lightningdesignsystem.com/design-tokens/)
- [Adobe](https://spectrum.adobe.com/page/design-tokens/)
- [Duet](https://www.duetds.com/tokens/)

Reply:
> There's only one headline per article

Except when there [isn't](https://www.theguardian.com/technology/2020/jan/20/jabra-elite-75t-review-small-and-long-lasting-airpod-beaters) 🙂

I'm happy to use `text.headlinePrimary` where `headline` is the **component**. I derived from what you said here:

> the _design system component_ to which this token is applied

that this was meant to refer to specific components that exist in the design system. If not, and it won't cause confusion, let's follow the same conventions?

Reply:
Yep that sounds fine to me! I'd be interested to hear what people think of the ergonomics of splitting imports on CSS property

```js
import { text } from 'editorialPalette'

const headline = css`
  color: ${text.headlinePrimary(format)};
`

const standfirst = css`
  color: ${text.standfirstPrimary(format)};
`
```

vs

```js
import { textHeadline, textStandfirst } from 'editorialPalette'

const headline = css`
  color: ${textHeadline(format)};
`

const standfirst = css`
  color: ${textStandfirst(format)};
`
```

The second API is more verbose but perhaps more explicit? It would be so cool if you could do away with all those function calls, but I understand it might become difficult to maintain.

```js
import { palette } from 'editorialPalette'

const { text } = palette(format)

const headline = css`
  color: ${text.headlinePrimary};
`

const standfirst = css`
  color: ${text.standfirstPrimary};
`
```

Reply:
In your second example wouldn't you still need a reference to `primary`?

I don't have a strong preference for one of those APIs over the other. I like two things about both:

- They take `Format` and give you back one or more colours.
- You can consume individual colours, _or_ retrieve the entire palette as an object. I think this will be important for flexibility (see [this comment](https://github.com/guardian/apps-rendering/pull/313#discussion_r421421338)).

Reply:
> In your second example wouldn't you still need a reference to primary?

Yep, sorry, an omission.

Cool. You're right, it's probably much of a muchness and you can potentially provide both APIs for flexibility

```js
import { p } from 'palette'
const { text } = p(format)

//...
css`
  color: ${text};
`
```

```js
import { text } from 'palette'

// ...
css`
  color: ${text(format)};
`
```

Reply:
We seem to have a few ways we could define the API based upon what you've suggested above 🤔

1. Object with all the colours for a given **property**, with keys for every **component**-**variant** combination.
```ts
text(format).headlinePrimary
```

2. Object with all the colours for a given **property**, with keys for every **component** that give you objects with all the **variant**s
```ts
text(format).headline.primary
```

3. An object for each **property**-**component** combination, where the keys of each object are the **variant**s
```ts
textHeadline(format).primary
```

4. Function for every **property**-**component**-**variant** combination, which return a single hex colour as a `string`
```ts
textHeadlinePrimary(format)
```

Reply:
Can we also consider:

5. A function for the entire **palette** that returns an object with each **property**, each of which has keys for every **component-variant** combination

```js
palette(format).text.headlinePrimary
```

?

Reply:
Oh sorry, yes that's also an API we'd want. With the above I was asking _which one_ of those four we should go with for individual colour lookups. I think we'd want the `palette` API to exist _in addition_ to that?

Reply:
My preference would be option 1. It's the same API as Source and I kinda like the ergonomics. Also there's some value in consistency as people would be able to switch between the Source palette and the Editorial palette without having to think too much.

Reply:
Ok, sounds good, I'm happy with the API.

My only concern is that we'd be generating a lot of unused colours with every call to the `text` function. For example, if we use it in the `Headline` component, we're only going to pull out one or two colours. We could solve this with memoisation, or by dependency injecting the full palette as DCR are planning to do.

I wonder if a _fifth_ option that looks a lot like (1) would be preferable:

```ts
text.headlinePrimary(format)
```

Reply:
Oh gosh, yes. More performant and preserves the ergonomics. I'm on board 🚢

Reply:
Ok, I've updated the code to use this new API.